src/encoded/static/components/visualizations.js

```diff
@@ -496,6 +606,335 @@ export class ChartTable extends React.Component {
+            }),
+        }));
+    }
+
+    handleSearch(event) {
+        // Unsanitized search term entered by user for display
+        this.setState({ unsanitizedSearchTerm: event.target.value });
+        // Search term entered by the user
+        const filterVal = String(sanitizedString(event.target.value));
+        this.setState({ searchTerm: filterVal });
+        const filteredData = this.state.data.filter((d) => {
+            if (filterVal === '') {
+                return true;
+            }
+            if (d.biosample_ontology.term_name) {
+                return sanitizedString(mapChromatinNames(d.value)).includes(filterVal) || sanitizedString(d.biosample_ontology.term_name).includes(filterVal) ||
+                sanitizedString(d.biosample_ontology.organ_slims.join(', ')).includes(filterVal);
+            }
+            return sanitizedString(mapChromatinNames(d.value)).includes(filterVal);
+        });
+        const fakeFacets = Object.keys(this.state.chartData)
+            .reduce((obj, key) => {
+                obj[key] = 0;
+                return obj;
+            }, {});
+        const newSelectedStates = [];
+        filteredData.forEach((d) => {
+            fakeFacets[mapChromatinNames(d.value)] += 1;
+            const chromatinValue = sanitizedString(mapChromatinNames(d.value));
+            if (!newSelectedStates.includes(chromatinValue)) {
+                newSelectedStates.push(chromatinValue);
+            }
+        });
+        this.setState({
+            filteredData,
+            filteredChartData: fakeFacets,
+            selectedStates: newSelectedStates,
+        });
+    }
+
+    clearSearch() {
+        // clear both search terms
+        this.setState(prevState => ({
+            searchTerm: '',
+            unsanitizedSearchTerm: '',
+            filteredData: prevState.data.filter(d => (prevState.selectedStates.includes(sanitizedString(mapChromatinNames(d.value))))),
+            filteredChartData: prevState.chartData,
+        }));
+    }
+
+    drawCharts() {
+        const data = this.props.data;
+        const initialChartData = Object.assign({}, initializedChromatinObject);
+        data.forEach((d) => {
+            initialChartData[mapChromatinNames(d.value)] += 1;
+        });
+        const chartData = Object.keys(initialChartData)
+            .filter(key => initialChartData[key] > 0)
+            .reduce((obj, key) => {
+                obj[key] = initialChartData[key];
+                return obj;
+            }, {});
+        const chartKeys = Object.keys(chartData);
+        const chartArray = chartKeys.map(key => chartData[key]);
+        const chartMax = Math.max(...chartArray);
+        let selectedStates = [];
+        for (let idx = 0; idx < chartKeys.length; idx += 1) {
+            if (chartData[chartKeys[idx]] > 0) {
+                selectedStates = [sanitizedString(chartKeys[idx])];
+                break;
+            }
+        }
```

Comment:
This looks like a chartKeys.find()? Then sanitize the result and put it in a single-element array? Or is that not the same?

Reply:
I think we need to leave it like this. Here we are initializing the selected state to be most active state for which there are results. Often that will be "Active TSS" but not always. Since we don't know what the selected state should be, we can't use `.find`.